code/modules/admin/permissionedit.dm

```diff
@@ -44,86 +264,120 @@ ADMIN_VERB(edit_admin_permissions, R_PERMISSIONS, "Permissions Panel", "Edit adm
-			output += "</tr>"
-		output += "</table></div><div id='top'><b>Search:</b> <input type='text' id='filter' value='' style='width:70%;' onkeyup='updateSearch();'></div></body>"
+		// We're gonna sanity check admin rank info, yea?
+		// We're pulling from the db as a source of truth here, instead of trusting the local lists
+		// First, pull all admin ranks which are used
+		var/datum/db_query/query_extract_admins = SSdbcore.NewQuery("SELECT IFNULL((SELECT byond_key FROM [format_table_name("player")] WHERE [format_table_name("player")].ckey = [format_table_name("admin")].ckey), ckey), [format_table_name("admin")].`rank` FROM [format_table_name("admin")]")
+		if(!query_extract_admins.warn_execute())
+			qdel(query_extract_admins)
+			return
+		var/list/admins_by_rank = list()
+		while(query_extract_admins.NextRow())
+			var/admin_key = query_extract_admins.item[1]
+			var/admin_rank = query_extract_admins.item[2]
+			for(var/datum/admin_rank/composed_rank as anything in ranks_from_rank_name(admin_rank))
+				admins_by_rank[composed_rank.name] += list(admin_key)
+		QDEL_NULL(query_extract_admins)
+
+		// Then, pull the full list of DB ranks to purity check against
+		var/datum/db_query/query_extract_ranks = SSdbcore.NewQuery("SELECT rank, flags, exclude_flags, can_edit_flags FROM [format_table_name("admin_ranks")]")
+		if(!query_extract_ranks.warn_execute())
+			qdel(query_extract_ranks)
+			return
+		var/list/all_db_ranks = list()
+		while(query_extract_ranks.NextRow())
+			var/loaded_rank = query_extract_ranks.item[1]
+			var/rank_flags = query_extract_ranks.item[2]
+			var/rank_exclude_flags = query_extract_ranks.item[3]
+			var/rank_can_edit_flags = query_extract_ranks.item[4]
+			all_db_ranks[loaded_rank] = list("rank" = loaded_rank, "flags" = rank_flags, "exclude_flags" = rank_exclude_flags, "can_edit_flags" = rank_can_edit_flags)
+		QDEL_NULL(query_extract_ranks)
+
+		output += "<h3>Admin ckeys with invalid ranks</h3>"
+		var/list/invalid_admin_ranks = admins_by_rank - all_db_ranks
+		if(length(invalid_admin_ranks)) // Add a nice leader to the output
+			output += "<hr style='background:#000000; border:0; height:1px'>"
+		else
+			output += "No invalid ranks found!"
+
+		for(var/illegal_rank in invalid_admin_ranks)
+			for(var/misapplied_admin in admins_by_rank[illegal_rank])
+				output += {"
+					[misapplied_admin] has the non-existent rank [illegal_rank] |
+					<a href='byond://?_src_=holder;[HrefToken()];editrightsbrowserhousekeep=1;editrightschange=[misapplied_admin]'>\[Change Rank\]</a> |
+					<a href='byond://?_src_=holder;[HrefToken()];editrightsbrowserhousekeep=1;editrightsremove=[misapplied_admin]'>\[Remove\]</a>
+					<hr style='background:#000000; border:0; height:1px'>
+				"}
+
+		output += "<h3>Unused DB Ranks</h3>"
+		var/list/unused_ranks = all_db_ranks - admins_by_rank
+		if(length(unused_ranks))
+			output += "<hr style='background:#000000; border:0; height:1px'>"
+		else
+			output += "No unused ranks found!"
```

Comment:
```suggestion
			output += "No unused ranks found."
```

This one would probably be more important to have as a neutral statement in case people get concerned they need an unused rank for some reason